compiler/plugins/target/AMD-AIE/aie/AMDAIEDmaToNpu.cpp

```diff
@@ -512,33 +511,37 @@ struct AMDAIEDmaToNpuPass : mlir::OperationPass<DeviceOp> {
 
     instructions[2] = count;
     instructions[3] = instructions.size() * sizeof(uint32_t);
-    ArrayRef<uint32_t> instsArrRef(instructions.data(), instructions.size());
-    device->setAttr(
-        "npu_instructions",
-        DenseUI32ResourceElementsAttr::get(
-            RankedTensorType::get(
-                instsArrRef.size(),
-                IntegerType::get(&getContext(), 32, IntegerType::Unsigned)),
-            "npu_instructions",
-            HeapAsmResourceBlob::allocateAndCopyInferAlign(instsArrRef)));
-    // The LX instructions for the entry point function are already generated by
-    // the pass hence we can safely delete the function as it is of no use to
-    // us. A reason to do this is that otherwise it is unceseccarily lowered to
-    // llvm where it can have a chance to crash in case the argument list is not
-    // lowerable for reasons such as memref's with dynamic offsets.
-    auto symName = dyn_cast_or_null<StringAttr>(device->getAttr("sym_name"));
+
+    // Note on use of 'DenseIntElementsAttr' below:
+    // We cannot use DenseUI32ResourceElementsAttr because it doesn't
+    // serialize-deserialize, and we (currently) serialize the module before
```

Comment:
I'm gonna have to take a closer look at this tomorrow but I don't understand what you mean by 

> we (currently) serialize the module before running aie2xclbin.

since looking at `aie2xclbin` I see it's still `ModuleOp moduleOp` and so I don't understand why we can't keep using `DenseUI32ResourceElementsAttr`

Reply:
Gets saved to disk here: https://github.com/nod-ai/iree-amd-aie/blob/9109a1fcae55c8f2f5a2c934e9f57b3a6b1b9967/compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp#L349

Gets loaded here: https://github.com/nod-ai/iree-amd-aie/blob/9109a1fcae55c8f2f5a2c934e9f57b3a6b1b9967/compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp#L378

It's not clear why it's written to file at all, I'll see if I can change that. 

To the point of issues with DenseUI32ResourceElementsAttr being serialized though, I see this in the serialized file `npu_instructions = dense_resource<npu_instructions> : tensor<380xui32>` but the actual data is nowhere to be seen. I've played around with the flags `OpPrintingFlags` used, no luck though. 

But I'll dig into why it's serialized at all.

Reply:
It's not at the bottom in the resource block...?

Reply:
> It's not at the bottom in the resource block...? Also do you have `elide-resources>10` turn on somewhere?

It's nowhere, and the default flags should work (I've also tried with higher elide options) BUT I've just tried without doing the serialize->deserialize and it works. Serialization is crust from when aie2xclbin was in iree-amd-aie, we don't need it

Reply:
![image](https://github.com/user-attachments/assets/3ca4ddca-9486-4534-8141-ee0ec665f835)